specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+    pub reference_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+/// Offers details on the  metadata.
+pub trait MultiTokenMetadataProvider {
+    fn mt_metadata(&self, token_id: TokenId) -> MultiTokenMetadata;
+}
+```
+# Drawbacks
+[drawbacks]: #drawbacks
+Doing this adds another spec and codebase to the standards. It could be seen that we could leave this to developers to implement custom solutions and have them create a token that just uses FT and NFT together. There is some additional complexity
+in ux, when considering batch size request and gas limitations, that might trip some developers up. 
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+The rationale for this design, is that we want to support developers that need batch requests. In many ecosystems, a single NFT or 
+series of NFT are not sufficient for representing and managing potentially 100s of tokens. Managing cross contract calls and the token contracts themselves become a huge burden on the developer. This will reduce the complexity, and allow developers to easily bridge over to other contracts like ERC-1155 on other chains that allow for a this style of representation.
+
+In the design phase it was considered to simply rely on the underlying implementations of fungible tokens and non-fungible tokens
+to be the scope of interaction with the chain. Doing this we would have tied the implementations to the bounds of FT and NFT. By loosing this up a bit, we are able to be a bit more flexible in what's possible.
+
+Not doing this means we really won't have a great way of supporting use cases where developers need to represent and manage
+large varying quantities and types of tokens. They will all have to implement it on their own, and that would make for a fragmented
+and weak ecosystem. Where every developer would not be able to reliably trade these assets.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+The unresolved questions, are really what type of metadata , is required for this spec?
+- We decided that having metadata on chain is good, and that at later time we can have a spec for extra data if needed
+
+Can we represent events in this spec, which would improve the ecosystem quite a bit? If we represent events what should those events be?
+- We decided events aren't here yet in the ecosystem, in the way we' like them to be so they are't apart of this standard
+
+Should we have a spec for TokenType?
+- We decided TokenType shouldn't be exposed for the public interface consumption. We will leave this description in the hands of the implementers
+
+Should we have a spec for offchain metadata?
+- We decided no spec for offchain metadata yet but maybe in the future
+
+Does the current storage management scheme work for people?
+- The current storage management scheme works for folks
```

Comment:
I would expect storage management would need to be more complex, because my storage requirements would go up for each FungibleToken type that I hold a balance for -- is that not the case here?  It seems fundamentally different than our existing case where the storage cost is stable/consistent

Reply:
Storage costs should be stable and consistent, for instance in the spike ahead implementation I did you can pick a max token_id size  , and a representation for tokens that you'd like to be fungible in nature, this costs stays constant I set a prefix of 64bit key space for BorshKeyStorage management. The difference in cost happens between or at least in estimation with the type of token. So FungibleTokens are more expensive to represent than non FungibleTokens in architecture, within my sort of large thought about the way to a standard core implementation might look. 

https://github.com/shipsgold/multi-token-standard-impl/pull/1

The result is I believe a stable storage cost estimation for whatever you're trying to create.